### DISCUSSION TITLE
How do I know if egui handles an event?

### DISCUSSION THREAD

**Discussion by asny** [emilk/egui#192] [Q&A] [answered]
Hi @emilk 

First of all, I think egui is a fantastic project so thanks for all of your hard work 💪 

I am doing an integration into my pet project [three-d](https://github.com/asny/three-d) and I almost got it working good enough for a release. However, there is one issue I can't figure out how to solve. In addition to gui, I also have a 3D scene with controls based on input events. Right now both egui and the 3D controls get the same events which means that if you drag a slider in the gui you also rotate the 3D model. So, how do I know if egui has handled an event and I therefore should not handle it in 3D?

Best regards, Asger.

**Comment by emilk** [comment, marked as answer]
Hi, and thanks for the kind words!

You can use `ctx.is_using_pointer()` and `ctx.wants_keyboard_input()`:
https://docs.rs/egui/0.9.0/egui/struct.Context.html#method.is_using_pointer 
https://docs.rs/egui/0.9.0/egui/struct.Context.html#method.wants_keyboard_input

**Reply by asny** [reply]
Arh, perfect. Thanks a lot 👍
